pints/_nested/__init__.py

```diff
@@ -50,12 +230,424 @@ def __init__(self, log_likelihood, log_prior):
+                    )
+                    self._sampler._accept_count += 1
+                    i_iter_complete = 1
+            if i_iter_complete == 0:
+                # Propose new samples
+                proposed = self._sampler.ask(self._n_workers)
+                # Evaluate their fit
+                if self._n_workers > 1:
+                    log_likelihood = self._evaluator.evaluate(proposed)
+                else:
+                    log_likelihood = self._evaluator.evaluate([proposed])[0]
+                sample, winners = self._sampler.tell(log_likelihood)
+                while sample is None:
+                    proposed = self._sampler.ask(self._n_workers)
+                    if self._n_workers > 1:
+                        log_likelihood = (              # pragma: no cover
+                            self._evaluator.evaluate(proposed))
+                    else:
+                        log_likelihood = self._evaluator.evaluate(
+                            [proposed])[0]
+                    sample, winners = self._sampler.tell(log_likelihood)
+                if winners.size > 0:
+                    if i_winners == 0:
+                        m_previous_winners = winners
+                        i_winners = 1
+                    else:
+                        m_previous_winners = [m_previous_winners, winners]
+                        m_previous_winners = np.concatenate(m_previous_winners)
+
+            # Check whether within convergence threshold
+            if i > 2:
+                self._diff_marginal_likelihood(i, d)
+                if (np.abs(self._diff) <
+                   self._marginal_log_likelihood_threshold):
+                    if self._log_to_screen:
+                        print(                              # pragma: no cover
+                            'Convergence obtained with Delta_z = ' +
+                            str(self._diff))
+
+                    # shorten arrays according to current iteration
+                    self._iterations = i
+                    self._v_log_Z = self._v_log_Z[0:(self._iterations + 1)]
+                    self._w = self._w[0:(
+                        self._n_active_points + self._iterations)]
+                    self._X = self._X[0:(self._iterations + 1)]
+                    self._m_inactive = self._m_inactive[0:self._iterations, :]
+                    break
+
+            # Show progress
+            self._update_logger()
+
+        # Calculate log_evidence and uncertainty
+        self._log_Z = self.marginal_log_likelihood()
+        self._log_Z_sd = self.marginal_log_likelihood_standard_deviation()
+
+        # Draw samples from posterior
+        n = self._posterior_samples
+        self._m_posterior_samples = self.sample_from_posterior(n)
+
+        return self._m_posterior_samples
+
+    def sample_from_posterior(self, posterior_samples):
+        """
+        Draws posterior samples based on nested sampling run.
```

Comment:
Is this to be called after it's finished running?

Reply:
It automatically calls in the run() method but can also be called afterwards if the user wants a new sample. I've added a comment to make this clearer.